target/consensource-cli
# Fix assertion CLI command

## Proposed change/fix

- [x] Add `INGESTION` org typ to org creation commmand
- [x] Implement `factory` subcommand for assertion creation
- [ ] ~~Implement `certificate` subcommand for assertion creation~~
- [ ] ~~Implement `standard` subcommand for assertion creation~~

**Update**: in the interest of time, I will leave the latter two subcommands unimplemented.

## Types of changes

What types of changes does this pull request introduce to ConsenSource? _Put an `x` in the boxes that apply_


## How to run/test

1. Clone down the [consensource-compose](https://github.com/target/consensource-compose) repo:
```bash
$ git clone --recurse-submodules https://github.com/target/consensource-compose.git
```
2. Make sure each submodule is up-to-date with the latest code changes
2.5 Checkout this branch for CLI submodule
3. Run `cargo update` for each submodule
4. Build a local image for this image, naming it `target/consensource-cli:local`
5. Build local images for other submodules (also tagged `:local`)
6. Run `docker-compose -f docker-compose.run-local.yaml up`
7. Exec into the CLI container
8. Create an agent
```bash
$ csrc agent create testagent --url http://api:9009
```
9. Create an ingestion organization
```bash
$ csrc organization create testingestionorg 4 ingestioncontactname 7632918628 JPN --city minneapolis --country usa --street_address '821 thor street' --url http://api:9009
```
10. Create a factory assertion
```
$ csrc assertion factory create <asserters_org_id> testfactoryassertion somename 6123456789 ENG --city ames --country usa --street_address '123 some st' --url http://api:9009
```
